Add download links to the download page

ap4y opened the pull request:
I have spent some time on  adding download links to the "Download" page and I'm curious what everyone think about this version:

| x86 | i686 | arm | nojs |
| ----- | ----- | --- | --- |
|  ![Screenshot_2020-10-06 Enter the void - Downloads](https://user-images.githubusercontent.com/673861/95148235-516afe80-07df-11eb-8c9f-fa8da1d4ca93.png)  |  ![Screenshot_2020-10-06 Enter the void - Downloads(1)](https://user-images.githubusercontent.com/673861/95148247-5b8cfd00-07df-11eb-8a25-0dc5aa396d60.png)  |  ![Screenshot_2020-10-06 Enter the void - Downloads(2)](https://user-images.githubusercontent.com/673861/95148256-621b7480-07df-11eb-9e45-05474789f2ba.png) | ![Screenshot_2020-10-06 Enter the void - Downloads(3)](https://user-images.githubusercontent.com/673861/95148270-68115580-07df-11eb-8a55-b3c30fa868bd.png) |

Notable changes:

- Pushed various archs to the tabs, I think users usually interested in the specific platform
- Added download links for all archs and flavors for live images and tarballs
- Tweaked details for each arch and pushed it to the side

The downside of these changes is that build date has to be updated in the config after each new build, I think it happens rarely so hopefully won't create too much load. If someone will notice that something is off, I'm happy to tweak it.

ericonr commented:
Hi! I really like these changes! I still have some thoughts on the text parts, but also a style consideration. How does this page look on a phone?

ap4y commented:
Thanks @ericonr. Mobile previews:

- [main archs](https://user-images.githubusercontent.com/673861/95041705-f6c89880-0733-11eb-8b85-5eea03a1c3a7.png) 
- [arm](https://user-images.githubusercontent.com/673861/95041718-fd571000-0733-11eb-806b-d16dbeb464a2.png)

ahesford commented:
I think this looks very good, but have a couple of suggestions that I think will improve the viewing experience:
1. Don't split the `x86_64*` tabs if you aren't also splitting the ARM tabs by C library. Put the glibc and musl links side-by-side in one list.
2. In the ARM tab, break the "live image" and rootfs options onto two lines, so you have something like "Live image, rootfs <br> Live image (musl), rootfs (musl)".

ericonr commented on _platforms/arm.md:
```diff
@@ -0,0 +1,23 @@
+{% capture download_details %}
+Live images can be written onto an SD card (i.e. using `dd`) and they allow you to have a ready to boot system. These images are prepared for 2GB SD cards. Alternatively, use the rootfs tarballs if you want to customize the partitions and filesystems.
+
+Connect to it in virtual terminal or via ssh and log in as `root`, password `voidlinux`.
```
`Connect to it with in a virtual terminal`

ericonr commented on _includes/download_pc.html:
```diff
@@ -0,0 +1,37 @@
+    <h3>{{ page.name }}</h3>
+    {{ include.content | markdownify }}
+    <div class="alert alert-warning" role="alert">
+      To install the desktop environment, DON’T choose “install from network”
+      choose the local install.
```
`DON'T choose "install from network", choose the local install.`

ericonr commented on _platforms/arm.md:
```diff
@@ -0,0 +1,23 @@
+- [USB Armory](https://wiki.voidlinux.org/USB_Armory)
+{% endcapture %}
+
+{% include download_arm.html content=download_details %}
```
I think we are losing some of the information here, such as the link to the page about RPis in the handbook. I also wouldn't like to lose the device information for each board (like `ARMv7, hard float` we currently have). Not sure if it could fit inside this, though.

ap4y replied:
I have added arch back without `hard float` since all devices are `hard float`. I have removed a link to the RPI docs because the copy before it points to the related doc page (`General and platform specific instructions for these images are available in the documentation.`), I think this should be enough in this case. WDYT?

ericonr replied:
Sounds great.

Chocimier commented on _layouts/download.html:
```diff
@@ -0,0 +1,20 @@
+
+  <div id="tab-content">
+    {% for platform in site.platforms %}
+    <div class="hidden">{{ platform.content }}</div>
```
Make sure that all content is available if js don't run.

ap4y replied:
Added additional separator behind `noscript` and added screenshot for the nojs variant.

Chocimier reviewed (commented):
I like appearance. Could you set up live demo to play with (eg. github pages), rather than screenshots?

ap4y commented:
Thanks for the reviews. I have pushed updates based on the feedback and updated screenshots in the first message. The only thing I haven't done is splitting "live images" and "rootfs" into separate lines, I tried different variants but felt like the links look the best inline. I have re-grouped links and added `glibc` and `musl` badges to make the separation a bit more clear, please let me know what you think about this :)

@Chocimier  you can test these changes locally, just pull the branch and  then  `bundle install` followed by `bundle exec jekyll s`, preview should be up on port `4000` by default.

ericonr commented:
I will try to set something up on GitHub Pages so we can discuss this more easily. I really want to get this design into the download pages.

ap4y commented:
I have deployed to GH pages from my branch. You can preview it here: https://ap4y.github.io/void-linux.github.io/download/

ericonr commented:
What do you think about taking the opportunity to increase the font size a bit? I always felt it was quite small

ap4y commented:
@ericonr are you referring to site-wide font updates or on the download page specifically? if it's the former then perhaps it will be better to handle this in  a separate PR along with other potential typography improvements? I personally think that light fonts are not a great option for the content heavy websites, maybe worth trying to switch to a regular font weight as quick fix? 14px font size that is being used is not small but in combination with certain fonts it might be harder to read.

ericonr commented:
For some reason the fonts only bother me on the download page, even though it seems to be the same style in the main page... I'm not sure why that is, but perhaps increasing it site wide wouldn't be too bad.

paper42 commented:
The font size is set in bootstrap.css, a best practice is not forcing users to a specific font size, so we should remove it.

Duncaen commented:
What about putting the text left and the download links right? This would put headlines on the left without JS enabled, would have the headline first in mobile.
The tab bar could have links with ids that scroll to the right architecture headline without JS.

ap4y commented:
Updated per @Duncaen's suggestions:

- Swapped panels on each tab
- Added tab links and header anchors

Updated  version is up on the same link.

the-maldridge commented on _config.yml:
```diff
@@ -11,3 +11,10 @@ snow:        0
+    output: true
+
+download_mirror: https://alpha.de.repo.voidlinux.org/live/current
+build_date: 2019-11-09
```
Please change to download_build_date for consistency.

the-maldridge commented on _includes/download_details_pc.md:
```diff
@@ -0,0 +1,7 @@
+Installable live images support a local installation (with the included packages) or a network installation (packages are downloaded from official repository).
+
+Log in as `anon`/`root`, password `voidlinux`.
```
In other places in our documentation we refer to user/pass combos with the slash, and this could be confusing.  Please revise to `anon/voidlinux`, `root/voidlinux`.

the-maldridge commented on _includes/download_details_pc.md:
```diff
@@ -0,0 +1,7 @@
+
+Log in as `anon`/`root`, password `voidlinux`.
+
+To start the installer just execute the `void-installer` utility with enough permissions (i.e., `sudo`).
```
s/enough/appropriate/

the-maldridge commented on _includes/download_details_pc.md:
```diff
@@ -0,0 +1,7 @@
+
+To start the installer just execute the `void-installer` utility with enough permissions (i.e., `sudo`).
+
+These images need at least 256 or 512 MB of RAM in order to work correctly.
```
Drop this line, it is unqualified and somewhat confusing.

the-maldridge commented on assets/js/misc.js:
```diff
@@ -0,0 +1,24 @@
+window.addEventListener("DOMContentLoaded", () => {
```
this file exclusively handles tabBars, so it should have a name that reflects that.

the-maldridge commented on _platforms/arm.md:
```diff
@@ -0,0 +1,37 @@
+---
```
I'm not entirely happy calling this arm when we have both specific dialects as well as platform filesystems.  I'd almost rather split this section into two and have one that just talks about that you can get arbitrary rootfs tarballs, and another page that talks about SBCs and links to the relevant pages in the documentation.

This would also be nice as I'd like to add a tab for the various cloud platforms that Void supports.

ap4y replied:
I have split ARM into 2 sections and sketched content for both, would appreciate a review on those.

re cloud images: I only see GCP tarball at https://alpha.de.repo.voidlinux.org/live/current/ , what other platforms we want to show on the downloads  page?

the-maldridge replied:
Both GCP and DigitalOcean are known to work and are some semblance of supported (Void itself has infrastructure running on DigitalOcean).  For both though the images are not distributed by default, so just having a place to link back to the docs on how to generate the image on demand would be great.

the-maldridge reviewed (changes requested):
Several nits and I'd like to split up the page that's currently called 'arm' into 2 pages with one being a page on rootfs's and one specifically talking about SBCs.

the-maldridge commented:
Looks pretty good.  All my major concerns have been addressed.   @void-linux/webmasters for final approval.  The cloud page can be added with just a pointer to mklive for now, and more instructions can be added later.

ericonr commented on _includes/download_details_pc.md:
```diff
@@ -0,0 +1,5 @@
+Installable live images support a local installation (with the included packages) or a network installation (packages are downloaded from official repository).
+
+Log in as `anon/voidlinux`, `root/voidlinux`.
```
```suggestion
You can log into these images as `anon` or `root`, and the password is `voidlinux`.
```

ericonr commented on _includes/download_details_pc.md:
```diff
@@ -0,0 +1,5 @@
+
+Log in as `anon/voidlinux`, `root/voidlinux`.
+
+To start the installer just execute the `void-installer` utility with appropriate permissions (i.e., `sudo`).
```
```suggestion
To start the installer, execute the `void-installer` utility with appropriate permissions (i.e., `sudo void-installer`).
```

ericonr commented on _platforms/arm.md:
```diff
@@ -0,0 +1,15 @@
+ROOTFS tarballs can be extracted to a previously prepared partition scheme or
+used for chroot installation.
+
+General specific instructions are available [in the
+documentation](https://docs.voidlinux.org/installation/guides/arm-devices/index.html).
+{% endcapture %}
```
```suggestion
General  and platform specific instructions are available [in the
documentation](https://docs.voidlinux.org/installation/guides/arm-devices/index.html).
{% endcapture %}
```

ericonr commented on _platforms/sbc.md:
```diff
@@ -0,0 +1,36 @@
+{% capture download_details %}
+Live images can be written onto an SD card (i.e. using `dd`) and they allow you to have a ready to boot system. These images are prepared for 2GB SD cards. Alternatively, use the rootfs tarballs if you want to customize the partitions and filesystems.
+
+Connect to it using a virtual terminal or via ssh and log in as `root/voidlinux`.
```
```suggestion
Connect to the system using a virtual terminal or SSH and log in as `root` with password `voidlinux`.
```

ericonr commented on _platforms/sbc.md:
```diff
@@ -0,0 +1,36 @@
+---
+
+{% capture download_details %}
+Live images can be written onto an SD card (i.e. using `dd`) and they allow you to have a ready to boot system. These images are prepared for 2GB SD cards. Alternatively, use the rootfs tarballs if you want to customize the partitions and filesystems.
```
```suggestion
Live images can be written onto an SD card (i.e. using `dd`) and they provide you with a ready to boot system. These images are prepared for 2GB SD cards. Alternatively, use the ROOTFS tarballs if you want to customize the partitions and filesystems.
```

Could `ROOTFS tarballs` be a link to the `arm` page, perhaps?

ericonr commented on _includes/download_pc.html:
```diff
@@ -0,0 +1,75 @@
+    <h3 id="{{ page.name }}">{{ page.name }}</h3>
+    {{ include.content | markdownify }}
+    <div class="alert alert-warning" role="alert">
+      To install the desktop environment, DON'T choose "install from network"
+      choose the local install.
```
```suggestion
      To install the packages for the desktop environment, DON'T choose "install from network",
      choose the "local install" option.
```

ericonr reviewed (changes requested):
Ok, I think I went through all the issues I noticed.

Man, I really like this page, thanks a lot!

the-maldridge commented:
@ericonr's feedback is good, once that's complete this will be merged.

ap4y commented:
Thanks for the feedback @ericonr, PR was updated with the fixes.

the-maldridge commented:
@ap4y please squash your changes to 1 commit, once you have done this ping me for merge.

ap4y commented:
@the-maldridge done
